Src/PChecker/CheckerCore/SystematicTesting/ControlledRuntime.cs

```diff
@@ -457,7 +460,14 @@ private StateMachine CreateStateMachine(StateMachineId id, Type type, string nam
 
             if (id is null)
             {
-                id = new StateMachineId(type, name, this);
+                if (name != null)
+                {
+                    id = new StateMachineId(type, name.Substring(2), this);
```

Comment:
What is this?

Reply:
This is because the new state machine's `InterfaceName` couldn't be passed in through `InitializeParametersEvent` anymore. I decided to pass it in is through the `name` parameters in CreateStateMachine() since it's not used for anything else. This function is originally getting called like this:
`var machineId = CreateStateMachine(createMachine, createdInterface.Substring(2), new InitializeParametersEvent(new InitializeParameters(createdInterface, payload)));` 
Now it gets called like this:
`var machineId = CreateStateMachine(createMachine, createdInterface, GetConstructorEvent(payload));`